scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/SchedulingMethodImpl.java

```diff
@@ -196,115 +197,135 @@ public int schedule() {
 
             // ask the policy all the tasks to be schedule according to the jobs list.
 
-            LinkedList<EligibleTaskDescriptor> taskRetrievedFromPolicy = currentPolicy.getOrderedTasks(descriptors);
+            LinkedList<EligibleTaskDescriptor> fullListOfTaskRetrievedFromPolicy = currentPolicy.getOrderedTasks(descriptors);
 
             //if there is no task to scheduled, return
-            if (taskRetrievedFromPolicy == null || taskRetrievedFromPolicy.isEmpty()) {
+            if (fullListOfTaskRetrievedFromPolicy == null || fullListOfTaskRetrievedFromPolicy.isEmpty()) {
                 return numberOfTaskStarted;
             }
 
-            if (logger.isDebugEnabled()) {
-                logger.debug("eligible tasks : " + taskRetrievedFromPolicy);
-            }
-
             schedulingService.unlockJobsToSchedule(toUnlock.values());
             toUnlock = null;
 
-            updateVariablesForTasksToSchedule(jobMap, taskRetrievedFromPolicy);
+            int subTasksListFrom = 0;
+            int subTasksListTo = Math.min(fullListOfTaskRetrievedFromPolicy.size(),
+                                          subTasksListFrom + freeResources.size());
+            LinkedList<EligibleTaskDescriptor> taskRetrievedFromPolicy;
 
-            for (EligibleTaskDescriptor etd : taskRetrievedFromPolicy) {
-                // load and Initialize the executable container
-                loadAndInit(((EligibleTaskDescriptorImpl) etd).getInternal());
-            }
+            while (subTasksListFrom < subTasksListTo) {
+                taskRetrievedFromPolicy = new LinkedList<>(fullListOfTaskRetrievedFromPolicy.subList(subTasksListFrom,
+                                                                                                     subTasksListTo));
```

Comment:
You could put the following logging in a separate method.

Reply:
OK